worlds/stardew_valley/__init__.py

```diff
@@ -171,15 +170,22 @@ def create_items(self):
                                for location in self.multiworld.get_locations(self.player)
                                if location.address is not None])
 
-        created_items = create_items(self.create_item, self.delete_item, locations_count, items_to_exclude, self.options, self.content,
-                                     self.random)
+        created_items = create_items(self.create_item, locations_count, items_to_exclude, self.options, self.content, self.random)
 
         self.multiworld.itempool += created_items
 
         setup_early_items(self.multiworld, self.options, self.player, self.random)
         self.setup_player_events()
         self.setup_victory()
 
+        # This is really a best-effort to get the total progression items count. The actual total can be impacted by the start_inventory_from_pool when items
+        # are removed from the bool, and probably when plando is applied as well... But since this is only used to spread grindy items across spheres, it's not
+        # a big deal if it's not 100% accurate.
```

Comment:
Three things:
1. `bool` -> `pool`
2. FWIW `start_inventory_from_pool` *is* being accounted for here, since they're made pre-collected long before here and Stardew's `get_filler_item_name` doesn't create additional progression items.
3. This is not only used for spreading items around, both the allsanity and perfection goals use it. This *is* a bug, and creates unwinnable seeds. For example: Plando'ing Blueberry Seeds onto Blueberry Seeds Shipsanity while also adding additional progression items through plando in a `minimal` world will result in a logically beatable, physically unbeatable world. This bug isn't really relevant to this code though as it already exists in Stardew

Reply:
> FWIW start_inventory_from_pool is being accounted for here, since they're made pre-collected long before here and Stardew's get_filler_item_name doesn't create additional progression items.

My understanding from reading Main.py is that they will be counted two times in the total progression items. They are pre-collected between `generate_early` and `create_regions`, but a duplicated version of the item will also be created in `create_items`. The total will not be decreased when the items are removed from the item pool. 

I might be wrong, tho. Would not be the first time I misunderstand what's happening in core.

Reply:
Oh then that's *also* a bug and isn't being accounted for and creates unwinnable seeds

Reply:
You're right about the plando bug, it can create unwinnable seeds. 

The simplest solution for both the plando and start inv issues seems to refresh the count of progression items in the `pre_fill` hook, but I'm not sure changing the total progression items mid-generation won't create other issues. It will need to be in another pr so it can be tested more thoroughly.

Reply:
A short term solution could be to block generation of Allsanity and Perfection goals with minimal accessibility.

Reply:
```suggestion
        # This is really a best-effort to get the total progression items count. The actual total can be impacted by the start_inventory_from_pool when items
        # are removed from the pool, and probably when plando is applied as well... But since this is only used to spread grindy items across spheres, it's not
        # a big deal if it's not 100% accurate.
```

Reply:
> A short term solution could be to prevent generation of Allsanity and Perfection goals with minimal accessibility. All progression items need to be collected to win, so it does not make that much sense to generate with minimal accessibility.

That's such a good hotfix, wow. I love it

Reply:
Nvm that start inventory from pool bug, it does not exist. Stardew does not support the option. We made the default start inventory option automatically remove items from the pool. Removed items are replaced by resource packs and accounted for properly because they are removed in the `create_items` method before the progression item count.